src/bundler.ts

```diff
@@ -102,346 +97,268 @@ export class Bundler {
-   * If the URL is part of the bundle, this method returns `true`.
-   *
-   * If the URL is part of another bundle, this method returns the url of that
-   * bundle.
-   *
-   * If the URL isn't part of a bundle, this method returns `false`
+   * @param {Array<string>} entrypoints The list of entrypoints that will be
+   *     analyzed for dependencies. The results of the analysis will be passed
+   *     to the `strategy`.
+   * @param {BundleStrategy} strategy The strategy used to construct the
+   *     output bundles. See 'polymer-analyzer/src/bundle-manifest'.
+   * @param {BundleUrlMapper} mapper A function that produces urls for the
+   *     generated bundles. See 'polymer-analyzer/src/bundle-manifest'.
    */
-  resolveBundleUrl(
-      url: string,
-      bundle: AssignedBundle,
-      manifest: BundleManifest): boolean|string {
-    const targetBundle = manifest.getBundleForFile(url);
-    if (!targetBundle || !targetBundle.url) {
-      return false;
-    }
-    if (targetBundle.url !== bundle.url) {
-      const relative = urlUtils.relativeUrl(bundle.url, targetBundle.url);
-      if (!relative) {
-        throw new Error('Unable to compute relative url to bundle');
-      }
-      return relative;
+  async bundle(
+      entrypoints: string[],
+      strategy?: BundleStrategy,
+      mapper?: BundleUrlMapper): Promise<DocumentCollection> {
+    // TODO(usergenic): Expose the generateBundleManifest operation
+    // as public and make the interface for bundle method take a manifest.
+    const manifest =
+        await this._generateBundleManifest(entrypoints, strategy, mapper);
+
+    const bundledDocuments: DocumentCollection =
+        new Map<string, BundledDocument>();
+
+    for (const bundleEntry of manifest.bundles) {
+      const bundleUrl = bundleEntry[0];
+      const bundle = {url: bundleUrl, bundle: bundleEntry[1]};
+      const bundledAst =
+          await this._bundleDocument(bundle, manifest, bundle.bundle.files);
+      bundledDocuments.set(
+          bundleUrl, {ast: bundledAst, files: Array.from(bundle.bundle.files)});
     }
-    return true;
+
+    return bundledDocuments;
   }
 
-  isStripExcludedHref(url: string): boolean {
-    if (!this.stripExcludes) {
-      return false;
+  /**
+   * Add HTML Import elements for each file in the bundle.  We append all the
+   * imports in the case any were moved into the bundle by the strategy.
+   * While this will almost always yield duplicate imports, they will be
+   * cleaned up through deduplication during the import phase.
+   */
+  private _appendHtmlImportsForBundle(
+      document: ASTNode,
+      bundle: AssignedBundle) {
+    for (const importUrl of bundle.bundle.files) {
+      const newUrl = urlUtils.relativeUrl(bundle.url, importUrl);
+      if (!newUrl) {
+        continue;
+      }
+      this._appendHtmlImport(this._findOrCreateHiddenDiv(document), newUrl);
     }
-    return this.stripExcludes.some(r => url.search(r) >= 0);
+    return document;
```

Comment:
1. It wasn't used.
2. It doesn't even match the type signature for the function.
3. I've removed it.
4. Thanks!